src/client.js

```diff
@@ -42,15 +42,43 @@ export default class Client {
     }, options);
 
     if (options.apiHostname && isString(options.apiHostname)) {
-      const apiHostnameParsed = url.parse(options.apiHostname);
-      options.apiProtocol = apiHostnameParsed.protocol || 'https:';
-      options.apiHost = apiHostnameParsed.host;
+      let apiHostname = options.apiHostname;
+
+      if (/^https?:\/\//i.test(apiHostname) === false) {
```

Comment:
@thomasconner there's a slight difference between this and the previous behavior. In the new case, if someone uses `http://` as the prefix, we will attach `https://` in front of it. This seems like it may confuse developers. Can we use the parsed url like before to figure out the `apiProtocol`?